consensus/polybft/contractsapi/gen2/main.go

```diff
@@ -0,0 +1,347 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"go/format"
+	"io/ioutil"
+	"strconv"
+	"strings"
+	"text/template"
+	"unicode"
+
+	gensc "github.com/0xPolygon/polygon-edge/consensus/polybft/contractsapi"
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/contractsapi/artifact"
+	"github.com/umbracle/ethgo/abi"
+)
+
+const (
+	contractVariableName   = "%sContract"
+	contractStructName     = "%sContractImpl"
+	contractVariableFormat = "%s = &%s{Artifact: %s}"
+	abiTypeNameFormat      = "var %sABIType = abi.MustNewType(\"%s\")"
+	eventNameFormat        = "%sEvent"
+)
+
+func main() {
+	cases := []struct {
+		contractName string
+		artifact     *artifact.Artifact
+		methods      []string
+		events       []string
+	}{
+		{
+			"StateReceiver",
+			gensc.StateReceiver,
+			[]string{
+				"commit",
+				"execute",
+			},
+			[]string{
+				"StateSyncResult",
+				"NewCommitment",
+			},
+		},
+		{
+			"ChildValidatorSet",
+			gensc.ChildValidatorSet,
+			[]string{
+				"commitEpoch",
+			},
+			[]string{},
+		},
+		{
+			"StateSender",
+			gensc.StateSender,
+			[]string{
+				"syncState",
+			},
+			[]string{
+				"StateSynced",
+			},
+		},
+		{
+			"CheckpointManager",
+			gensc.CheckpointManager,
+			[]string{
+				"submit",
+			},
+			[]string{},
+		},
+	}
+
+	rr := render{}
+
+	res := []string{}
+	globalVariables := []string{}
+
+	for _, c := range cases {
+		tmpl := `type {{.StructName}} struct {
+			Artifact *artifact.Artifact
+
+			{{.Fields}}
+		}`
+
+		genContractFields := func(fields []string) string {
+			structFields := make([]string, len(fields))
+
+			for i := 0; i < len(fields); i++ {
+				title := strings.Title(fields[i])
+				structFields[i] = fmt.Sprintf("%s %s", title, title)
+			}
+
+			return strings.Join(structFields, "\n")
+		}
+
+		inputs := map[string]interface{}{
+			"StructName": fmt.Sprintf(contractStructName, firstCharacterLowercase(c.contractName)),
+			"Fields":     genContractFields(c.methods),
+		}
+
+		globalVariables = append(globalVariables, fmt.Sprintf(contractVariableFormat,
+			fmt.Sprintf(contractVariableName, c.contractName), inputs["StructName"], c.contractName))
+
+		contractStruct := renderTmpl(tmpl, inputs)
+
+		res = append(res, contractStruct)
+
+		for _, method := range c.methods {
+			res = append(res, rr.GenMethod(c.contractName, c.artifact.Abi.Methods[method]))
+		}
+
+		for _, event := range c.events {
+			res = append(res, rr.GenEvent(c.artifact.Abi.Events[event]))
```

Comment:
I believe here we should lowercase the first character...?

Reply:
Fix: https://github.com/0xPolygon/polygon-edge/pull/1062/commits/b5b79ab945f3a4be11047cf3ea7261782bfc177b